### DISCUSSION TITLE
Commentary panel

### DISCUSSION THREAD

**Discussion by tobias-klein** [ezra-bible-app/ezra-bible-app#890] [Ideas]
Hello everyone,

I am working on support for commentary modules (see #886) and visualizing Bible commentaries for the currently selected verse in a new commentary panel.

It currently looks like this:
![image](https://user-images.githubusercontent.com/17739895/226097846-a5eec5be-43e3-44cb-8410-752b75a2f07c.png)

Do you have any feedback or wishes regarding the functionality around Bible commentaries?

**Comment by Citizenyolo** [comment]
Hi Tobias,

Firstly, thank you for your service by making this great Bible Study tool available to everyone!

My wish list regarding commentary is as follows:

1. Would be great to set the font size. I use a large resolution monitor and the commentary text is too tiny to read. I can adjust the Bible text size but not the commentary (at least I have not figured out a way to do so).
2. Can we please have Dark Mode as optional rather than setting it automatically. Would be great to be able to switch it on/off/mirror OS setting as an option.
3. I can place the commentary section either to the left or to the bottom in relation to the Bible Text. Most people find more natural reading from left to right. Would be great to allow the Bible text to be on the left side and the commentary on the right side.
4. Can you pls enable copying selected parts of commentary to the clipboard. A lot of people (myself included) use external note apps (Onenote or Mac Notes) and/or Translation apps like DeepL. It would make much easier to use these apps with Copy/Paste being enabled for commentary text.

Thanks for listening and keep up the great work!
Zoltan

**Reply by tobias-klein** [reply]
Hi @Citizenyolo
1) There is an option since 1.12 (released last November) to also adjust the font size of the side panel along with the Bible text when changing the font size. So that should be working for you now based on the latest two releases available.
2) Dark Mode is currently only mirroring the OS setting on macOS. I suppose you are on a Mac? We could indeed add an option to turn off that behavior and make it configurable like on other platforms. I added an issue for this enhancement request: #1042 
3) That would be a more extensive change to the layout concept of the UI. Unfortunately, I currently do not have the capacity to implement such a change.
4) That is a good point. I just added an issue for this enhancement request: #1041

**Comment by ariasrodolfo** [comment]
There must be a selection box for each comment module, and each comment must be displayed only for the module that is selected, not all at the same time, because there are comments with a lot of text, and a person can have dozens of comment modules installed. it is my opinion.

**Reply by tobias-klein** [reply]
Hi @ariasrodolfo 
Thanks for your feedback. I would consider this an advanced configuration option for the commentary side panel. I just added an issue: #1043

**Comment by MikeM65** [comment]
Thank-you for the commentary section. I started using Ezra Bible a few weeks ago and the commentary section is one of my favorite parts. I very much like how all commentaries are listed right there. If you could make it so there is a pop-up, or can start a new window with verses listed in the commentary, it would be icing on the cake. Like if I hover the mouse over a reference mentioned in the commentary. This would be especially nice for the Treasury of Scripture Knowledge commentary that is like a chain reference.

**Reply by tobias-klein** [reply]
Hi @MikeM65 
Thanks for your feedback and your suggestion. I just added a new issue for your request:
#1044

**Comment by Citizenyolo** [comment]
Hi Tobias,

Happy New Year!

I just checked the latest version and I'm very impressed with the inclusion of the TSK. Thanks so much!

I have two suggestions on the Commentary panel to help reduce the time having to scroll through all open commentaries to get to the information I'm looking for.

**1. Remember the Commentary collapse / expand status by book**

I normally check the TSK and 3 favorite commentaries to get a quick idea. I would ideally leave them open (the little button with the arrow pointing down) and keep the other books closed (arrow pointing to the right). If I need more info I can alway collapse 1-2 more commentaries of my choice. Currently I MUST scroll through EVERYTHING to get to the information I'm looking for.

Would be great if it always offered books open or closed as I left them by default (or enable it as an option in settings) even when I close and reopen the app.

**2. Enable change of orders of commentary books**

This would also help get to the info I need quicker. I downloaded the TSK as the last book on my list. In my Bible study workflow this would be however the first book on my list, so I now I start by having to scroll through all the commentaries to get there. Would be great to be able to change the order of the books displayed on the list. Either some Up/Down arrow appearing to move up or down or with a drag and drop solution, whatever solution would be easy to implement.

**+1 backup on "hover-over"**

Finally, I also support MikeM65's suggestion to enable a a "hover-over" support in the commentary section (or if feasible across the whole app). I love the solution implemented to open the chain of references as a new tab  to analyze further. However oftentimes when I read about a reference in a commentary it would just like to hover over and see the verse referenced in a window popping up and when I move the cursor away, the window disappearing so I can carry over reading . It would just save a lots of unnecessary mouse clicks to open  and close and carry on reading the commentary.

As always, thanks for listening and for your service!

**Reply by tobias-klein** [reply]
Hi Zoltan!

Happy new year to you too! :) 

> **1. Remember the Commentary collapse / expand status by book**
> 
> I normally check the TSK and 3 favorite commentaries to get a quick idea. I would ideally leave them open (the little button with the arrow pointing down) and keep the other books closed (arrow pointing to the right). If I need more info I can alway collapse 1-2 more commentaries of my choice. Currently I MUST scroll through EVERYTHING to get to the information I'm looking for.
> 
> Would be great if it always offered books open or closed as I left them by default (or enable it as an option in settings) even when I close and reopen the app.

Good suggestion!

Just to be sure about your request, when you say *by book* you mean by *commentary book* and not by Bible book, right?

> **2. Enable change of orders of commentary books**
> 
> This would also help get to the info I need quicker. I downloaded the TSK as the last book on my list. In my Bible study workflow this would be however the first book on my list, so I now I start by having to scroll through all the commentaries to get there. Would be great to be able to change the order of the books displayed on the list. Either some Up/Down arrow appearing to move up or down or with a drag and drop solution, whatever solution would be easy to implement.

Great suggestion, I can imagine to implement this! I just created issue #1182 for this.

> 
> **+1 backup on "hover-over"**
> 
> Finally, I also support MikeM65's suggestion to enable a a "hover-over" support in the commentary section (or if feasible across the whole app). I love the solution implemented to open the chain of references as a new tab to analyze further. However oftentimes when I read about a reference in a commentary it would just like to hover over and see the verse referenced in a window popping up and when I move the cursor away, the window disappearing so I can carry over reading . It would just save a lots of unnecessary mouse clicks to open and close and carry on reading the commentary.
> 

Thanks for sharing your thoughts. This would be a desktop-specific optimization. Generally, I try to design the user interface in such a way that it works both on desktop and mobile and therefore I try to avoid mouse over based actions, because it limits the use to desktop. I see your point, though. I'll think about it.

**Reply by Citizenyolo** [reply]
Yes, by book (to expand or collapse) I meant the commentary only. 

I didn't realize there was a mobile app, too. I tried looking at the App Store For "Ezra Bible" but got no hit. Is it already available or WIP?

**Reply by tobias-klein** [reply]
Thanks for clarifying!
The app is only available for Android, not for iOS. You find it in the [Google Play Store](https://play.google.com/store/apps/details?id=net.ezrabibleapp.cordova).
